resources/monitor.rb

```diff
@@ -53,10 +57,10 @@
   end
 end
 
-def yaml_dir
+def config_file_path(resource_name)
   if node['datadog']['agent6']
-    ::File.join(node['datadog']['agent6_config_dir'], 'conf.d')
+    ::File.join(node['datadog']['agent6_config_dir'], 'conf.d', "#{resource_name}.d")
   else
-    ::File.join(node['datadog']['config_dir'], 'conf.d')
+    ::File.join(node['datadog']['config_dir'], 'conf.d', "#{resource_name}.d")
```

Comment:
Agent v5 doesn't support the per-check subdirectories, i.e. for integration `foo` Agent 5 will only read the config file in `<agent_config_dir>/conf.d/foo.yaml`.

So we'll need some logic specific to Agent 5, see what was done in #558 (and let's acknowledge and thank the initial contributor as well :) )

Reply:
My bad for the unique file for agent 5. It's done with 28cf699.